### PR TITLE
feat: implement JWT token management system

### PR DESCRIPTION
**Pull request by KatnessChen**
🎯 **Overview**
---------------

This PR implements a JWT token management system for the transaction tracker backend, providing secure authentication and authorization capabilities with database-backed token lifecycle management.

📋 **Summary of Changes**
-------------------------

### **Core Features Implemented:**

-   ✅ **JWT Token Lifecycle Management** - Complete token creation, validation, revocation, and cleanup
-   ✅ **Email-based Authentication** - User signup endpoint with comprehensive validation
-   ✅ **Composite Hash Security** - Enhanced token security with unique hash generation
-   ✅ **Database Token Storage** - Persistent token management with soft delete support
-   ✅ **Authentication Middleware** - Request authentication and error handling
-   ✅ **Comprehensive Test Suite** - Unit and integration tests for all components

#### **Models & Database:**

-   `internal/models/jwt_token.go` - JWT token model with lifecycle fields
-   `migrations/003_create_jwt_tokens_table.sql` - JWT tokens table creation
-   `migrations/004_add_deleted_at_to_jwt_tokens.sql` - Soft delete support

#### **Services & Repositories:**

-   `internal/services/jwt_service.go` - JWT token business logic and validation
-   `internal/repositories/jwt_repository.go` - Database operations for JWT tokens
-   `internal/repositories/user_repository.go` - User-related database operations

#### **API Layer:**

-   `api/handlers/auth.go` - Authentication endpoints including user signup
-   `api/middlewares/auth.go` - JWT authentication middleware
-   Updated `api/routes/routes.go` - New authentication routes

### **API Endpoints:**

**`POST /api/v1/signup`:**

-   Email uniqueness validation
-   Password confirmation matching
-   Input validation (email format, password strength)
-   User creation with bcrypt password hashing

**`POST /api/v1/login`**

- ENHANCED with proper JWT token lifecycle management (not just token generation)

**`POST /api/v1/logout`:**

-   Revokes the current JWT token
-   Database-backed token invalidation

**`POST /api/v1/logout-all`:**

-   Revokes all JWT tokens for the current user
-   Useful for security scenarios (password change, etc.)

**`POST /api/v1/refresh-token`:**

-   Generates new JWT token using existing valid token
-   Maintains session continuity

**`GET /api/v1/me`:**

-   Returns current user profile information
-   Shows active token count for the user

#### **Testing:**

-   `test/jwt_service_test.go` - JWT service layer tests (15 test cases)
-   `jwt_repository_test.go` - Repository layer tests (15 test cases)
-   `auth_middleware_test.go` - Middleware tests (11 test cases)

🔧 **Technical Details**
------------------------

### **Security Features:**

-   **Token Hashing**: Composite hash generation for enhanced security
-   **Input Validation**: Comprehensive validation for all authentication endpoints
-   **Password Security**: Bcrypt hashing with proper salt rounds
-   **Token Expiration**: Configurable token expiration (24h default)
-   **Token Revocation**: Database-backed token revocation support

### **Database Schema:**

jwt_tokens table:

- id (primary key)

- user_id (foreign key)

- token_hash (unique, indexed)

- expires_at (indexed for cleanup)

- device_info (JSON)

- last_used_at (activity tracking)

- revoked_at (soft delete)

- created_at, updated_at, deleted_at

🧪 **Testing Coverage**
-----------------------

### **Test Statistics:**

-   **Total Test Cases**: 41+ comprehensive tests
-   **Test Categories**: Unit tests, integration tests, error handling
-   **Coverage Areas**: Service layer, repository layer, middleware layer
-   **Validation Tests**: Input validation, edge cases, error scenarios

### **Test Highlights:**

-   Token generation and validation workflows
-   Database operations and transaction handling
-   Authentication middleware request processing
-   Error handling and edge case scenarios
-   Password hashing and validation logic

### CODE REVIEW

File: backend/internal/models/jwt_token.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+type JWTToken struct {
+	BaseModel
+	ID         string     `gorm:"type:varchar(36);primaryKey" json:"id"`
+	UserID     uint       `gorm:"not null;index:idx_user_tokens" json:"user_id"`
+	TokenHash  string     `gorm:"type:varchar(255);not null;uniqueIndex" json:"token_hash"`
+	IssuedAt   time.Time  `gorm:"not null;default:CURRENT_TIMESTAMP" json:"issued_at"`
+	ExpiresAt  time.Time  `gorm:"not null" json:"expires_at"`
+	RevokedAt  *time.Time `gorm:"null" json:"revoked_at,omitempty"`
+	LastUsedAt *time.Time `gorm:"null" json:"last_used_at,omitempty"`
+	DeviceInfo string     `gorm:"type:json;null" json:"device_info,omitempty"`
```
Comment (KatnessChen) [inline backend/internal/models/jwt_token.go:17]: Property `DeleteAt` seems to be missing
